ci-operator/step-registry/upi/install/azurestack/upi-install-azurestack-commands.sh

```diff
@@ -0,0 +1,206 @@
+  exit 1
+fi
+
+# Ensure ignition assets are configured with the correct invoker to track CI jobs.
+export OPENSHIFT_INSTALL_INVOKER="openshift-internal-ci/${JOB_NAME_SAFE}/${BUILD_ID}"
+export TEST_PROVIDER='azurestack'
+
+dir=/tmp/installer
+mkdir "${dir}"
+pushd "${dir}"
+cp -t "${dir}" \
+    "${SHARED_DIR}/install-config.yaml"
+
+if ! pip -V; then
+    echo "pip is not installed: installing"
+    if python -c "import sys; assert(sys.version_info >= (3,0))"; then
+      python -m ensurepip --user || easy_install --user 'pip'
+    else
+      echo "python < 3, installing pip<21"
+      python -m ensurepip --user || easy_install --user 'pip<21'
+    fi
+fi
+
+CLUSTER_NAME=$(python -c 'import yaml;data = yaml.full_load(open("install-config.yaml"));print(data["name"])')
+AAD_CLIENT_ID=$(jq -r .clientId ${SHARED_DIR}/osServicePrincipal.json)
+AZURE_REGION=ppe3
+SSH_KEY=$(python -c 'import yaml;data = yaml.full_load(open("install-config.yaml"));print(data["sshKey"])')
+BASE_DOMAIN=$(python -c 'import yaml;data = yaml.full_load(open("install-config.yaml"));print(data["baseDomain"])')
+TENANT_ID=$(jq -r .tenantId ${SHARED_DIR}/osServicePrincipal.json)
+AAD_CLIENT_SECRET=$(jq -r .clientSecret ${SHARED_DIR}/osServicePrincipal.json)
+APP_ID=$(jq -r .clientId ${SHARED_DIR}/osServicePrincipal.json)
+
+export CLUSTER_NAME
+export AAD_CLIENT_ID
+export AZURE_REGION
+export SSH_KEY
+export TENANT_ID
+export BASE_DOMAIN
+export AAD_CLIENT_SECRET
+export APP_ID
+
+az cloud register \
+    -n PPE \
+    --endpoint-resource-manager "https://management.ppe3.stackpoc.com" \
+    --suffix-storage-endpoint "ppe3.stackpoc.com" 
+az cloud set -n PPE
+az cloud update --profile 2019-03-01-hybrid
+az login --service-principal -u $APP_ID -p $AAD_CLIENT_SECRET --tenant $TENANT_ID > /dev/null
+
+# shellcheck disable=SC2034
+SUBSCRIPTION_ID=$(az account show | jq -r .id)
+export SUBSCRIPTION_ID
+export AZURE_AUTH_LOCATION="${SHARED_DIR}/osServicePrincipal.json"
+
+echo "Installing python modules: yaml"
+python3 -c "import yaml" || pip3 install --user pyyaml
+
+# remove workers from the install config so the mco won't try to create them
+python3 -c '
+import yaml;
+path = "install-config.yaml";
+data = yaml.full_load(open(path));
+data["compute"][0]["replicas"] = 0;
+open(path, "w").write(yaml.dump(data, default_flow_style=False))'
+
+openshift-install create manifests
+
+# we don't want to create any machine* objects 
+rm -f openshift/99_openshift-cluster-api_master-machines-*.yaml
+rm -f openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+
+oc adm release extract $RELEASE_IMAGE_LATEST --credentials-requests --cloud=azure
+files=$(ls *credentialsrequest*.yaml)
+for f in $files
+do
+  SECRET_NAME=$(python -c "import yaml;data = yaml.full_load(open('${f}'));print(data['name']['spec']['secretRef']['name'])")
+  SECRET_NAMESPACE=$(python -c "import yaml;data = yaml.full_load(open('${f}'));print(data['name']['spec']['secretRef']['namespace'])")
```

Comment:
you probably already noticed this, but this is causing the error:
```
~/installer /output
pip 20.3.4 from /usr/lib/python2.7/site-packages/pip (python 2.7)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'module' object has no attribute 'full_load'
```
because we're not editing the files it might be a bit more clear to just use `yq` here.
Something like:
```shell
    SECRET_NAME=$(yq r "$f" spec.secretRef.name)
    SECRET_NAMESPACE=$(yq  r "$f" spec.secretRef.namespace)
```

Reply:
That's just me using python instead of python3. I'll fix it.